core/src/main/java/apoc/trigger/TriggerNewProcedures.java

```diff
@@ -68,12 +69,15 @@ private void checkTargetDatabase(String databaseName) {
     @Procedure(name = "apoc.trigger.install", mode = Mode.WRITE)
     @Description("Eventually adds a trigger for a given database which is invoked when a successful transaction occurs.")
     public Stream<TriggerInfo> install(@Name("databaseName") String databaseName, @Name("name") String name, @Name("statement") String statement, @Name(value = "selector")  Map<String,Object> selector, @Name(value = "config", defaultValue = "{}") Map<String,Object> config) {
-        checkInSystemWriter();
-        checkTargetDatabase(databaseName);
-        Map<String,Object> params = (Map)config.getOrDefault("params", Collections.emptyMap());
-        
-        TriggerInfo triggerInfo = TriggerHandlerNewProcedures.install(databaseName, name, statement, selector, params);
-        return Stream.of(triggerInfo);
+
+        return withTransaction(tx -> {
+            checkInSystemWriter(tx);
+            checkTargetDatabase(databaseName, tx);
```

Comment:
For these two you don't need to create a new transaction, you can just recycle the one you are in, so I'd suggest reshaping it like:

```java
        checkInSystemWriter();
        checkTargetDatabase(databaseName);
        Map<String,Object> params = (Map)config.getOrDefault("params", Collections.emptyMap());

        return withTransaction(tx -> {
            TriggerInfo triggerInfo = TriggerHandlerNewProcedures.install(databaseName, name, statement, selector, params, tx);
            return Stream.of(triggerInfo);
        });
```

so you only open a new system transaction at the moment where you need it

Reply:
Changed 👍